### PR TITLE
WT-12837 update test/3rdparty packages

### PR DESCRIPTION
**Pull request by bzm-jas**
This PR updates third party packages to remove uses of distutils.

The following packages have had their versions updated:
python-subunit: 0.0.16 -> 1.4.4
testtools: 0.9.34 -> 2.7.1
and discover 0.4.0 has been modified to remove references to distutils

### CODE REVIEW

File: test/3rdparty/discover-0.4.0/README.txt [hunk 1]
```diff
@@ -92,6 +92,11 @@ http://code.google.com/p/unittest-ext/source/browse/trunk/discover.py
 CHANGELOG
 =========
 
+2024/04/16 0.4.0
```
Comment (ajmorton) [inline test/3rdparty/discover-0.4.0/README.txt:95]: Can you flag that this is a change was made by the WiredTiger team and doesn't come from the discover maintainers? Right now this looks like identical to the prior changelogs

File: test/3rdparty/discover-0.4.0/README.txt
Comment (ajmorton) [inline test/3rdparty/discover-0.4.0/README.txt:1]: Can we rename `discover-0.4.0` to `discover-0.4.0-locally-modified`? We did the same thing with `concurrencytest` to make it clear that our copy diverges from the original repo.

**Review by ajmorton — COMMENTED** [review]
Looking good! Two requests from me:

File: test/3rdparty/discover-0.4.0/README.txt
Comment (bzm-jas) [reply test/3rdparty/discover-0.4.0/README.txt:1]: renamed here: [50c70e3](https://github.com/wiredtiger/wiredtiger/pull/10491/commits/50c70e3e5ec16acfc5fcbaa9671a7fb87510e97a)

File: test/3rdparty/discover-0.4.0/README.txt (hunk 1 above)
Comment (bzm-jas) [reply test/3rdparty/discover-0.4.0/README.txt]: Sounds good, I've moved the WT changes to the start of the README file and added headers to make it clear. See changes here: [b869f07](https://github.com/wiredtiger/wiredtiger/pull/10491/commits/b869f07be7ac4b1a20f7864833f4abea449cb9be)
Please let me know if you think the current format is clear enough, happy to add additional notes if needed. :))

Comment (ajmorton) [reply test/3rdparty/discover-0.4.0/README.txt]: Looks perfect thanks!

**Review by ajmorton — APPROVED** [review]
LGTM!

**Review by jeremythorp — APPROVED** [review]
LGTM!

**Pull request merged**